### PR TITLE
Document how to use robotology-superbuild software with WSL2 

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/robotology/robotology-superbuild/issues/361 by documenting the use of `/etc/wsl.conf` for WSL2 that for sure it supports it, and leaving the old workaround for WSL1. 

Fix https://github.com/robotology/robotology-superbuild/issues/427 with an in-depth guide on how to configure X Servers with WSL2.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -313,14 +313,48 @@ If for any reason you do not want to use the provided scripts and you want to ma
...
+As all the software running on Linux distributions can run unmodified on Windows via WSL, to install the robotology-superbuild in WSL you can just install a Debian-based distribution for WSL, and then follow the instructions on how to install the [robotology-superbuild on Linux](#linux). As the WSL enviroment is nevertheless different, there are few things you need to care before using the robotology-superbuild on WSL, that are listed in the following, depending on whetever you are using WSL2 or WSL1 .  
 
-### Run graphical applications on WSL
+### WSL2
+
+#### Run graphical applications on WSL2
+The Linux instance in WSL2 are running as part of a lightweight virtual machine, so effectively the IP addresso of the WSL2 instance will be different from the IP address
```
Comment (pattacini) [inline README.md:323]: `address`

File: README.md [hunk 2]
```diff
@@ -313,14 +313,48 @@ If for any reason you do not want to use the provided scripts and you want to ma
...
 
-As all the software running on Linux distributions can run unmodified on Windows via WSL, to install the robotology-superbuild in WSL you can just install a Debian-based distribution for WSL, and then follow the instructions on how to install the [robotology-superbuild on Linux](#linux). As the WSL enviroment is nevertheless different, there are few things you need to care before using the robotology-superbuild on WSL, that are listed in the following.
+As all the software running on Linux distributions can run unmodified on Windows via WSL, to install the robotology-superbuild in WSL you can just install a Debian-based distribution for WSL, and then follow the instructions on how to install the [robotology-superbuild on Linux](#linux). As the WSL enviroment is nevertheless different, there are few things you need to care before using the robotology-superbuild on WSL, that are listed in the following, depending on whetever you are using WSL2 or WSL1 .
```
Comment (pattacini) [inline README.md:318]: a few things

File: README.md [hunk 3]
```diff
@@ -313,14 +313,48 @@ If for any reason you do not want to use the provided scripts and you want to ma
...
+first need to install an X Server for Windows. Furthermore, you will need to configure your application to connect to the X Server that is running on the Windows host, you can do 
+so by adding the following lines in the `~/.bashrc` file of the WSL2 instance:
+~~~
+export WINDOWS_HOST=$(grep nameserver /etc/resolv.conf | awk '{print $2}')
+export DISPLAY=${WINDOWS_HOST}:0.0
+~~~
+As unfortunately the IP addresses of the virtual IP network change at every reboot, it is also necessary to configure the X Server that you use to accept connection for arbitrary IP addresses. Check  [`doc/wsl2-xserver-configuration.md`](doc/wsl2-xserver-configuration.md) for instructions on how to do so on several X Servers.
+
+#### Sanitize PATH enviroment variable for WSL2 
+y default, the `PATH` enviroment variable in WSL will contain the path of the host Windows system, see https://github.com/microsoft/WSL/issues/1640 and https://github.com/microsoft/WSL/issues/1493. This can create problems,
```
Comment (pattacini) [inline README.md:334]: By default

**Review by pattacini — CHANGES_REQUESTED** [review]
A few minor typos to fix

**Comment by traversaro** [comment]
Thanks @pattacini , I fixed your comments in https://github.com/robotology/robotology-superbuild/pull/468/commits/8704086607888bd68f58ac90b35b51005adc3025 .

**Review by pattacini — APPROVED** [review]

**Pull request merged**